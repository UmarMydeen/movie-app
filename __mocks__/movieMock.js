const movieMock = {
    imdbID: 'mockimdbID1',
    Title: 'mockTitle',
    Poster: 'mockPoster',
    imdbRating: '9.2',
    Poster: 'mockUrl'
}

export default movieMock;