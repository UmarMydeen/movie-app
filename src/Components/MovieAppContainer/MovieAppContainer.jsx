import React from 'react';
import MovieFilterBar from '../MoviesFilterBar/MoviesFilterBar';
import MoviesGrid from '../MoviesGrid/MoviesGrid';

const MovieAppContainer = (props) => {
    return (
        <div className="container">
            <MovieFilterBar />
            <MoviesGrid />
        </div>
    );
};

export default MovieAppContainer;