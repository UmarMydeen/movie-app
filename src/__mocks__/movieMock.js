const movieMock = {
    imdbID: 'mockimdbID1',
    Title: 'mockTitle',
    Poster: 'mockPoster'
}

export default movieMock;