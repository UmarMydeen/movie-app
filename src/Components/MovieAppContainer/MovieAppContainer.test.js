import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore } from "redux";
import { TestableMovieAppContainer as MovieAppContainer } from './MovieAppContainer';
import movieMock from '../../../__mocks__/movieMock';
import rootReducer from '../../Reducers/index';

let mockProps;

const initialState = {
    user: { name: "mock name" },
};

const store = createStore(rootReducer, initialState);

beforeEach(() => {
    mockProps = {
        getMovieList: jest.fn(),
    }
});

test('should render component correctly', () => {
    const component = render(
        <Provider store={store}>
            <MovieAppContainer {...mockProps} />
        </Provider>
    );
    expect(component).toMatchSnapshot();
});