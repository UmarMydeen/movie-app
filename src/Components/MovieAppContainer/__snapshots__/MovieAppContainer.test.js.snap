// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="row mx-2"
        >
          <div
            class="col-3"
          >
            <div
              class="input-group input-group-sm mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                  id="inputGroup-sizing-sm"
                >
                  Search By Title
                </span>
              </div>
              <input
                aria-describedby="inputGroup-sizing-sm"
                aria-label="Small"
                class="form-control"
                type="text"
              />
            </div>
          </div>
          <div
            class="col-4"
          >
            <div
              class="input-group input-group-sm mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                  id="inputGroup-sizing-sm"
                >
                  Location
                </span>
              </div>
              <select
                aria-label="Default select example"
                class="form-select"
                data-testid="connectedDropdown"
              >
                <option
                  value=""
                >
                  --Select--
                </option>
              </select>
            </div>
          </div>
          <div
            class="col-4"
          >
            <div
              class="input-group input-group-sm mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                  id="inputGroup-sizing-sm"
                >
                  Language
                </span>
              </div>
              <select
                aria-label="Default select example"
                class="form-select"
                data-testid="connectedDropdown"
              >
                <option
                  value=""
                >
                  --Select--
                </option>
              </select>
            </div>
          </div>
          <div
            class="col-1"
          >
            <button
              class="btn btn-outline-secondary"
              type="button"
            >
              <svg
                class="bi bi-sort-alpha-down"
                fill="currentColor"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10.082 5.629 9.664 7H8.598l1.789-5.332h1.234L13.402 7h-1.12l-.419-1.371h-1.781zm1.57-.785L11 2.687h-.047l-.652 2.157h1.351z"
                  fill-rule="evenodd"
                />
                <path
                  d="M12.96 14H9.028v-.691l2.579-3.72v-.054H9.098v-.867h3.785v.691l-2.567 3.72v.054h2.645V14zM4.5 2.5a.5.5 0 0 0-1 0v9.793l-1.146-1.147a.5.5 0 0 0-.708.708l2 1.999.007.007a.497.497 0 0 0 .7-.006l2-2a.5.5 0 0 0-.707-.708L4.5 12.293V2.5z"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="row row-cols-6 row-cols-md-5"
        >
          <p>
             No Data Found
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="row mx-2"
      >
        <div
          class="col-3"
        >
          <div
            class="input-group input-group-sm mb-3"
          >
            <div
              class="input-group-prepend"
            >
              <span
                class="input-group-text"
                id="inputGroup-sizing-sm"
              >
                Search By Title
              </span>
            </div>
            <input
              aria-describedby="inputGroup-sizing-sm"
              aria-label="Small"
              class="form-control"
              type="text"
            />
          </div>
        </div>
        <div
          class="col-4"
        >
          <div
            class="input-group input-group-sm mb-3"
          >
            <div
              class="input-group-prepend"
            >
              <span
                class="input-group-text"
                id="inputGroup-sizing-sm"
              >
                Location
              </span>
            </div>
            <select
              aria-label="Default select example"
              class="form-select"
              data-testid="connectedDropdown"
            >
              <option
                value=""
              >
                --Select--
              </option>
            </select>
          </div>
        </div>
        <div
          class="col-4"
        >
          <div
            class="input-group input-group-sm mb-3"
          >
            <div
              class="input-group-prepend"
            >
              <span
                class="input-group-text"
                id="inputGroup-sizing-sm"
              >
                Language
              </span>
            </div>
            <select
              aria-label="Default select example"
              class="form-select"
              data-testid="connectedDropdown"
            >
              <option
                value=""
              >
                --Select--
              </option>
            </select>
          </div>
        </div>
        <div
          class="col-1"
        >
          <button
            class="btn btn-outline-secondary"
            type="button"
          >
            <svg
              class="bi bi-sort-alpha-down"
              fill="currentColor"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.082 5.629 9.664 7H8.598l1.789-5.332h1.234L13.402 7h-1.12l-.419-1.371h-1.781zm1.57-.785L11 2.687h-.047l-.652 2.157h1.351z"
                fill-rule="evenodd"
              />
              <path
                d="M12.96 14H9.028v-.691l2.579-3.72v-.054H9.098v-.867h3.785v.691l-2.567 3.72v.054h2.645V14zM4.5 2.5a.5.5 0 0 0-1 0v9.793l-1.146-1.147a.5.5 0 0 0-.708.708l2 1.999.007.007a.497.497 0 0 0 .7-.006l2-2a.5.5 0 0 0-.707-.708L4.5 12.293V2.5z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="row row-cols-6 row-cols-md-5"
      >
        <p>
           No Data Found
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
